src/lib/methods/method.ts

```diff
@@ -1,48 +1,13 @@
-import retry, { Options as RetryOptions } from 'p-retry';
 import { DynamoDB } from '../dynamodb';
 import { Table } from '../table';
 import { Params } from '../types';
-import { QueryBuilder } from '../types/query-builder';
-import { retryErrorHandler, configureRetryOptions } from '../utils';
+import { BaseMethod } from './base-method';
 
-export abstract class Method implements QueryBuilder {
+export abstract class Method extends BaseMethod {
```

Comment:
Is there any benefit of this? Since this does not have custom logic compared to the `BaseMethod`?

Reply:
This is mostly for Typings, as Method has a field of type Params while BatchMethod has a field of type BatchParams. This seemed like the cleanest solution without having to do major refactoring or breaking changes. Like this we keep all the shared logic together while we can separate the differing interface.